External/FEXCore/Source/Interface/HLE/Thunks/Thunks.cpp

```diff
@@ -56,6 +62,54 @@ namespace FEXCore {
           Thread->CTX->HandleCallback(Thread, (uintptr_t)callback);
         }
 
+        /**
+         * Instructs the Core to redirect calls to functions at the given
+         * address to another function. The original callee address is passed
+         * to the target function through an implicit argument stored in r11.
+         *
+         * The primary use case of this is ensuring that host function pointers
+         * returned from thunked APIs can safely be called by the guest.
+         */
+        static void LinkAddressToGuestFunction(void* argsv) {
+            struct args_t {
+                uintptr_t original_callee;
```

Comment:
I would name this something among the lines `guest_linked_address` or `guest_entrypoint_address` and `guest_function` or `guest_handler_function`. 

Keeping a tab on what is guest and what is host can get very confusing, so I like to be verbose there

Reply:
There is no requirement for the callee address to be a host function. While it's used like that in practice, I reckon thinking about it in terms of a callee being replaced by a target address is most useful here.